### PR TITLE
scalastyle 1.0.0

### PR DESCRIPTION
**Pull request by ericbn**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Also don't install the default_config resource, only use it for the test.
The scalastyle script code is as recommended at http://www.scalastyle.org/command-line.html

### CODE REVIEW

File: Formula/scalastyle.rb [hunk 1]
```diff
@@ -1,31 +1,21 @@
...
+  url "https://oss.sonatype.org/content/repositories/releases/org/scalastyle/scalastyle_2.12/1.0.0/scalastyle_2.12-1.0.0-batch.jar"
+  sha256 "e9dafd97be0d00f28c1e8bfcab951d0e5172b262a1d41da31d1fd65d615aedcb"
 
   bottle :unneeded
 
   resource "default_config" do
-    url "https://raw.githubusercontent.com/scalastyle/scalastyle/v0.8.0/lib/scalastyle_config.xml"
-    version "0.8.0"
-    sha256 "feb61c4a09373717f94d277487e8118be524bb4c3265eb6d2fdee7a1fa362e50"
+    url "https://raw.githubusercontent.com/scalastyle/scalastyle/v1.0.0/lib/scalastyle_config.xml"
```
Comment (ilovezfs) [inline Formula/scalastyle.rb]: `version "1.0.0"`

File: Formula/scalastyle.rb [hunk 2]
```diff
@@ -37,6 +27,8 @@ def main(args: Array[String]) {
         }
       }
     EOS
-    system bin/"scalastyle", testpath/"test.scala"
+    resource("default_config").stage do
```
Comment (ilovezfs) [inline Formula/scalastyle.rb]: `testpath.install resource("default_config")`

We don't need to stage it in a separate temp directory.

**Review by ilovezfs — DISMISSED** [review]
Please explain more about why don't need to provide the config file.

**Comment by ericbn** [comment]
@ilovezfs, I don't have any strong opinion about installing the default config with the theme. I just think that:
 * scalastyle script should not pass a `--config "#{etc}/scalastyle_config.xml"` parameter, as the idea of the default config is just to serve as a template for creating our own config, which even may vary between different projects. Always loading the default config may also load conflicting rules, when the "customized" config is also passed as parameter to the script...
 * the default config file should be removed when the scalastyle installation is removed.
 * as the default config varies between scalastyle versions, I'm not sure if the file should be installed to `/usr/local/etc/` and kept there. Users also may be tempted to change the file there (as, again this is just a template config, intended to be edited to the user's liking), and an upgrade of scalastyle could override that with a newer version, which I don't think is a good approach.

Thoughts?

**Comment by apjanke** [comment]
If it's just supposed to serve as an example and isn't required for functionality, that does sound like `etc` is the wrong place for it. It might be nice to provide it in `share` under the formula installation so people could use it as a reference. Though if it's not provided as part of the distribution's releases, we should maybe lean toward not including it, since Homebrew tries to stick closely to distributions as provided by upstream.

**Comment by ericbn** [comment]
@apjanke, the upstream distribution does not include the example config file. See http://www.scalastyle.org/command-line.html

The distribution is just a executable `jar` file. And they provide a link to the documentation about the config:

> Run the jar like:
> ...
> or similar. You’ll need a [configuration](http://www.scalastyle.org/configuration.html).

@ilovezfs, updated the commit with your change requests.

**Comment by apjanke** [comment]
Yeah, that hardly counts as a "distribution". I'd lean toward including it, then.

**Comment by ericbn** [comment]
> Yeah, that hardly counts as a "distribution". I'd lean toward including it, then.

@apjanke, you mean "toward **not** including it [the default/sample configuration file]", right?

**Comment by apjanke** [comment]
No, I do mean "lean toward including it": since upstream hasn't put together a real distribution, we don't have an upstream distribution to stick closely to. That `.jar` file, while it's an archive file, is funcitonally just an executable binary.

But the best option IMHO would be to hassle upstream to provide a full installable distribution that includes all the necessary files; something that could be installed with a `make install` command or similar.

**Comment by ericbn** [comment]
@apjanke, the way this tools is used is:

 * There's an executable jar to run scalastyle in the command line. Actually there are also plugins to directly integrate scalastyle into build tools like [sbt](http://www.scalastyle.org/sbt.html) or [maven](http://www.scalastyle.org/maven.html) (in which the installation of the command-line scalastyle is not even needed).
 * There's a configuration file with the rules that scalastyle should use when doing the static analysis of a specific project code. The configuration file can be anywhere, as there's no enforcement by scalastyle on where it should be, we just need to give the path to it. 
* And the configuration file provided by scalastyle is just indented to be an example or a template. Developers or teams would generally edit it to their liking, and put the edited one in the place they think makes better sense for how they organize their projects, which I think would rarely be `/usr/local/etc/` or alongside the command-line scalastyle executable jar file...

So maybe a complete distribution of scalastyle command-line tool could contain an example configuration file. But I also think it makes sense to not have it, given the points above. Also, my [first comment](https://github.com/Homebrew/homebrew-core/pull/17414#issuecomment-326312495) is still valid to extend this one...

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ericbn** [comment]
@ilovezfs, any position on this?

To summarize, my changes are:
 1. Update from scala 2.11 to 2.12
 2. Update from scalastyle 0.8.0 to 1.0.0
 3. Remove the `--config` parameter from the [script command](https://github.com/Homebrew/homebrew-core/pull/17414/files#diff-c4c8acf2bbea191046346968c43d7b6cR19), as recommended by the [original documentation](http://www.scalastyle.org/command-line.html).
 4. Don't install the default configuration, and just use it for testing.

I think item 3 fixes an error in the actual script, because always loading the default configuration may conflict with a second customized configuration being passed as parameter later. And the usual way to work with scalastyle CLI is to always pass a custom configuration as parameter.

Now, I don't have strong opinions about item 4. The default configuration could be copied somewhere during this installation. Just mind that it will only serve as a reference for the actual customized configuration that users will create. Also, the [original configuration documentation](http://www.scalastyle.org/configuration.html) already has a link to the default configuration:

> ### Quick start
> You can just use the default Scalastyle configuration [scalastyle_config.xml](http://www.scalastyle.org/scalastyle_config.xml). Right-click this link and select `Save link as`. This is the one we use for Scalastyle development, so it’s obviously the best :-)

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
@ericbn this seems fine to me. You should tweak it to use `bin.write_jar_script` though.

**Comment by ericbn** [comment]
@ilovezfs, cool, just updated the commit to use `bin.write_jar_script`.

**Comment by ilovezfs** [comment]
Thanks again for another contribution to Homebrew, @ericbn!

**Pull request merged**